tests/js/common/shell/shell-index.js

```diff
@@ -55,6 +55,22 @@ function IndexSuite() {
       collection = null;
     },
 
+
+    testIndexProgress : function () {
+      var c = internal.db._create("c0l");
+      var docs = [];
+      for (var i = 0; i < 10000; ++i) {
+        docs.push({name : "name" + i});
+      }
+      c.insert(docs);
+      internal.debugSetFailAt("fillIndex::pause");
+      c.ensureIndex({ type: "persistent", fields:["name"]});
+      console.error("hu");
+      require('internal').sleep(1);
+      console.error("ha");
+      internal.debugSetFailAt("fillIndex::unpause");
```

Comment:
This is not a "real" test, in the sense of that does not have any conditions under which it will fail
So it executes something that one can attend and watch, but for automated tests it is not so helpful.
Apart from that, the test suite will be execute against a single server and cluster alike, and in the cluster it will likely not be any useful, because the failure points would need to be propagated to all DB servers for this (which I guess is not what is done here).